### PR TITLE
add a 'life of an entity' catalog article

### PR DESCRIPTION
**Pull request by freben**
Started poking at this during the weekend, don't feel I have the time to finish it right now but wanted to get whatever's there merged anyway since it is a fairly highly requested thing

### CODE REVIEW

File: docs/features/software-catalog/life-of-an-entity.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+change is detected. The LDAP provider is driven externally by a timer loop that
+occasionally triggers a full update. Some future provider may be entirely event
+driven, feeding off an event bus or web hook. There is no magic coordination
+among providers; if they need to arrange synchronization or locking among
+themselves for example to avoid duplicate work across multiple catalog service
+machines, they need to handle that out-of-band.
+
+The entities that are emitted get some coarse validation applied to them, to
+ensure that they adhere to the most basic rules of the system. They need to have
+a `kind`, a `metadata.name`, and optionally a `metadata.namespace`. Apart from
```
Comment (timbonicus) [inline docs/features/software-catalog/life-of-an-entity.md:97]: [apiVersion](https://backstage.io/docs/features/software-catalog/descriptor-format#common-to-all-kinds-the-envelope) too? Or is that not strictly required?

**Review by timbonicus — APPROVED** [review]
A beautiful serialization 🧠  => 📄

File: docs/features/software-catalog/life-of-an-entity.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+There are several stages involved in the processing.
+
+> TODO: More info here
+
+## Stitching
+
+The stitching is currently a fixed process, that cannot be modified or extended.
+This means that any modifications you want to make on the final result, has to
+happen during ingestion or processing.
+
```
Comment (jhaals) [inline docs/features/software-catalog/life-of-an-entity.md:126]: Perhaps adding something like this, edit as you wish:

Stitching finalises the entity by combining all data gathered such as relations, errors and the entity itself into a final object which is essentially the result from the catalog API. Entities that are no longer referenced by a location are annotated with `backstage.io/orphan: true` to indicate to humans and machines that this entity is no longer being updated.

File: docs/features/software-catalog/life-of-an-entity.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+The high level processes involved are:
+
+- _Ingestion_, where entity providers fetch raw entity data from external
+  sources and seed it into the database,
+- _Processing_, where the policies and processors continually treat the ingested
+  data and may emit both other raw entities (that are also subject to
+  processing), errors, relations to other entities, etc.,
+- _Stitching_, where all of the data emitted by various processors are stitched
+  together into the final output entity.
```
[the remark this answers is not in the document]
Comment (freben) [reply docs/features/software-catalog/life-of-an-entity.md:40]: Ah that's a good point, I'll try to make use of the word "assembled" and similar instead.

File: docs/features/software-catalog/life-of-an-entity.md (hunk 1 above)
Comment (freben) [reply docs/features/software-catalog/life-of-an-entity.md:97]: Yeah that's right, the entire envelope actually has to be valid. I'll rewrite a little.

File: docs/features/software-catalog/life-of-an-entity.md (hunk 2 above)
Comment (freben) [reply docs/features/software-catalog/life-of-an-entity.md:129]: Good formulation, will steal it in whole or partially in the next update

**Pull request merged**